### PR TITLE
Bump k8s deps for 1.21.0 release

### PR DESCRIPTION
**Pull request by sunjayBhatia**
See also https://github.com/projectcontour/contour/pull/3581

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
Seems blocked on https://github.com/kubernetes-sigs/controller-runtime/issues/1473

**Comment by youngnick** [comment]
Hmmm, I don't think we'll be able to do this for the 1.15 release then, which is unfortunate, since the controller is already using that client-go version.

**Comment by sunjayBhatia** [comment]
We could pick a main branch commit of controller runtime to use temporarily since it’s supposedly fixed there, what to people think?

**Comment by skriss** [comment]
> We could pick a main branch commit of controller runtime to use temporarily since it’s supposedly fixed there, what do people think? Would be good to keep the operator usable properly on k8s 1.21 so our compatibility matrix is correct

Wondering if we even need to bump the client-go version; I'd guess that things would work fine on a 1.21 cluster even without updating the deps. If we do want to pin to a main branch commit, we should probably eyeball what else is in there to make sure we're not shooting ourselves in the foot

**Comment by youngnick** [comment]
Given that the Operator is still alpha, we probably can get away with trusting to hope for the client-go compatibility for now.

**Comment by sunjayBhatia** [comment]
pulled https://github.com/projectcontour/contour-operator/pull/328 out so we can remove the dep bump from consideration

**Comment by danehans** [comment]
@sunjayBhatia can you resolve the merge conflict when you have a moment?

**Comment by sunjayBhatia** [comment]
> @sunjayBhatia can you resolve the merge conflict when you have a moment?

focusing on https://github.com/projectcontour/contour-operator/pull/328 for a bit since that is problematic with/without the client change

**Comment by skriss** [comment]
Saw there's a beta controller-runtime release out now: https://github.com/kubernetes-sigs/controller-runtime/tree/v0.9.0-beta.0

I'll leave this to @danehans to decide if he wants to take that dependency for 1.15, or wait for an official release.

File: go.mod
Comment (danehans) [inline go.mod:16]: @sunjayBhatia v0.9.x is for testing only. I think v0.8.3 would be a better choice, unless I'm missing something that's needed in 0.9.0.

**Review by danehans — CHANGES_REQUESTED** [review]

File: go.mod
Comment (sunjayBhatia) [reply go.mod:16]: This was done because of https://github.com/kubernetes-sigs/controller-runtime/issues/1473

Comment (sunjayBhatia) [reply go.mod:16]: Compilation fails with 0.8.3

Comment (danehans) [reply go.mod:16]: That's a good enough reason to go with v0.9.x.

**Comment by danehans** [comment]
> I'll leave this to @danehans to decide if he wants to take that dependency for 1.15, or wait for an official release.

I'm fine with alpha.

**Review by danehans — APPROVED** [review]

**Review by skriss — APPROVED** [review]
LGTM but I'd probably bump it to that new beta, just 'cause.

**Pull request merged**
